### PR TITLE
11 add playlist feature

### LINKED ISSUE #11: Add playlist feature
- Add playlist feature

### PR DESCRIPTION
**Pull request by rdshinde**
- playlist feature added

### CODE REVIEW

File: src/context/user-data-context/UserDataProvider.js [hunk 1]
```diff
@@ -56,6 +60,15 @@ const UserDataProvider = ({ children }) => {
         setWatchlaterVideos(serverResponse.data?.watchlater || []);
       } else if (serverResponse?.data?.history) {
         setHistoryVideos(serverResponse.data?.history || []);
+      } else if (serverResponse?.data?.playlists) {
+        setPlaylists(serverResponse.data?.playlists || []);
+      } else if (serverResponse?.data?.playlist) {
+        setPlaylists((prev) => [
```
Comment (nishasen) [inline src/context/user-data-context/UserDataProvider.js:66]: I like how you are using history here

**Review by nishasen — APPROVED** [review]
Neat and organized code, good naming convention

File: src/components/playlist-modal/PlaylistModal.jsx [hunk 2]
```diff
@@ -1,46 +1,104 @@
...
+      setPlaylistModal({ state: false, videoData: {} });
+    } else {
+      Toast({ type: "error", msg: "Please select playlist to add video." });
+    }
+  };
   return (
     <div
       className={`playlist-modal ${
-        !showPlaylistModal && "hide-playlist-modal"
+        !isPlaylistModal.state && "hide-playlist-modal"
```
Comment (krishanu2911) [inline src/components/playlist-modal/PlaylistModal.jsx:45]: Liked this part

**Review by krishanu2911 — APPROVED** [review]
Good to go no changes required Well written code 👍🏻

**Pull request merged**